tests/datasets/files_dataset_test.py

```diff
@@ -0,0 +1,150 @@
+    assert str(e_ctx.value) == "No benchmarks"
+
+
+def test_populated_dataset(populated_dataset: FilesDataset):
+    for _ in range(2):
+        assert list(populated_dataset.benchmark_uris()) == [
+            "benchmark://test-v0/e.txt",
+            "benchmark://test-v0/f.txt",
+            "benchmark://test-v0/g.jpg",
+            "benchmark://test-v0/a/a.txt",
+            "benchmark://test-v0/a/b.txt",
+            "benchmark://test-v0/b/a.txt",
+            "benchmark://test-v0/b/b.txt",
+            "benchmark://test-v0/b/c.txt",
+            "benchmark://test-v0/b/d.jpg",
+        ]
+        assert populated_dataset.n == 9
+
+
+def test_populated_dataset_benchmark_lookup(populated_dataset: FilesDataset):
+    bm = populated_dataset.benchmark("benchmark://test-v0/e.txt")
+    assert bm.uri == "benchmark://test-v0/e.txt"
+    assert bm.proto.uri == "benchmark://test-v0/e.txt"
+    assert bm.proto.program.uri == f"file:///{populated_dataset.dataset_root}/e.txt"
+
+
+def test_populated_dataset_first_file(populated_dataset: FilesDataset):
+    bm = next(populated_dataset.benchmarks())
+    assert bm.uri == "benchmark://test-v0/e.txt"
+    assert bm.proto.uri == "benchmark://test-v0/e.txt"
+    assert bm.proto.program.uri == f"file:///{populated_dataset.dataset_root}/e.txt"
+
+
+def test_populated_dataset_benchmark_lookup_not_found(populated_dataset: FilesDataset):
+    with pytest.raises(LookupError) as e_ctx:
+        populated_dataset.benchmark("benchmark://test-v0/not/a/file")
+
+    assert str(e_ctx.value).startswith(
+        "Benchmark not found: benchmark://test-v0/not/a/file"
+    )
+
+
+def test_populated_dataset_with_file_extension_filter(populated_dataset: FilesDataset):
+    populated_dataset.benchmark_file_suffix = ".jpg"
+    assert list(populated_dataset.benchmark_uris()) == [
+        "benchmark://test-v0/g",
+        "benchmark://test-v0/b/d",
+    ]
+    assert populated_dataset.n == 2
+
+
+@pytest.mark.parametrize(
+    "requested_uri", (None, "benchmark://test-v0", "benchmark://test-v0/")
+)
+def test_populated_dataset_random_benchmark(
+    populated_dataset: FilesDataset, requested_uri: Optional[str]
+):
+    populated_dataset.benchmark_file_suffix = ".jpg"
+
+    populated_dataset.seed(1)
+    benchmarks_a = [populated_dataset.benchmark(requested_uri).uri for _ in range(50)]
+    populated_dataset.seed(1)
+    benchmarks_b = [populated_dataset.benchmark(requested_uri).uri for _ in range(50)]
+    populated_dataset.seed(2)
+    benchmarks_c = [populated_dataset.benchmark(requested_uri).uri for _ in range(50)]
+
+    assert benchmarks_a == benchmarks_b
+    assert benchmarks_b != benchmarks_c
+
+    assert set(benchmarks_a) == set(populated_dataset.benchmark_uris())
+    assert set(benchmarks_c) == set(populated_dataset.benchmark_uris())
+
+
+def test_populated_dataset_get_benchmark_by_index(populated_dataset: FilesDataset):
+    for i in range(populated_dataset.n):
+        populated_dataset.get_benchmark_by_index(i)
+
```

Comment:
some assert seem to be missing here?

Reply:
Also seeing some out of range throws would be nice.

Reply:
Ahah yes looks like I forgot to add any tests :D Done